src/bp/nlu-core/entities/microsoft-extractor/index.ts

```diff
@@ -0,0 +1,380 @@
+      'Millimeter',
+      'Milímetro',
+      'Hectomètre',
+      'Décimètres',
+      'Miglio',
+      'Inch',
+      'Metro',
+      'Quilômetro',
+      'Milha',
+      'Nanometer',
+      'Milla',
+      'Kilómetro',
+      'Pulgada',
+      'Polegada',
+      'Pied',
+      'Kilomètres',
+      'Pouce',
+      'Piede',
+      'Chilometro'
+    ],
+    weight: [
+      'Ounce',
+      'Oncia',
+      'Libra',
+      'Metric ton',
+      'Gallon',
+      'Pound',
+      'Kilogram',
+      'Tonelada',
+      'Tonne métrique',
+      'Ton',
+      'Barrel',
+      'Tonelada métrica',
+      'Onza',
+      'Libbra',
+      'Gram',
+      'Tonnellata',
+      'Onça',
+      'Tonne',
+      'Livre'
+    ],
+    area: [
+      'Mètre carré',
+      'Metro cuadrado',
+      'Square decameter',
+      'Square hectometer',
+      'Square millimeter',
+      'Chilometro quadrato',
+      'Acro',
+      'Quilômetro quadrado',
+      'Square kilometer',
+      'Square meter',
+      'Kilómetro cuadrado',
+      'Square decimeter',
+      'Acre',
+      'Metro quadrato',
+      'Square centimeter',
+      'Metro quadrado',
+      'Hectare',
+      'Kilomètre carré'
+    ],
+    bits: ['Kilobyte', 'Terabyte', 'bit', 'Gigabyte', 'Megabit', 'Megabyte', 'Bit'],
+    speed: [
+      'Miles par heure',
+      'Kilometer per hour',
+      'Milla por hora',
+      'Meter per second',
+      "Miglia all'ora",
+      'Mile per hour',
+      'Milha por hora'
+    ]
+  }
+
+  private getDucklingUnitMappings = unit => {
+    for (const [type, subtypes] of Object.entries(this.ducklingUnitMapping)) {
+      if (subtypes.includes(unit)) {
+        return type
+      }
+    }
+  }
+
+  private ducklingDateMappings = {
+    // https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-datetimev2?tabs=1-3%2C2-1%2C3-1%2C4-1%2C5-1%2C6-1
+    'datetimeV2.datetimerange': 'duration',
+    'datetimeV2.daterange': 'duration',
+    'datetimeV2.timerange': 'duration',
+    'datetimeV2.duration': 'duration',
+    'datetimeV2.date': 'time',
+    'datetimeV2.datetime': 'time',
+    'datetimeV2.time': 'time'
+  }
+
+  private mapDucklingDates(entity): EntityExtractionResult {
+    const newEntity = { ...entity }
+    const newType = this.ducklingDateMappings[entity.type]
+    newEntity.type = newType
+    newEntity.entityId = `system.${newType}`
+
+    switch (newType) {
+      case 'duration':
+        let timeString = ''
+        if (newEntity.value.Mod) {
+          timeString += newEntity.value.Mod + '___'
+        }
+        if (newEntity.value.start) {
+          timeString += newEntity.value.start
+        }
+        if (newEntity.value.end) {
+          timeString += '___' + newEntity.value.end
+        }
+        newEntity.value = timeString
+        break
+
+      case 'time':
+        newEntity.value = newEntity.value.value
+        break
+
+      default:
+        this.logger?.error(`Type not mapped for ${JSON.stringify(newType, null, 2)}`)
```

Comment:
DANGER

1. We forgot about a date subtype in microsoft
2. The user writes a utterance with this exact type
3. We throw an error but the user can't do anything about it...

does this feel like a good UX to you?